File: leetcode/3001-4000/3101-3200/3101-3110/3105-longest-strictly-increasing-or-strictly-decreasing-subarray/LongestStrictlyIncreasingOrStrictlyDecreasingSubarray.ts
function longestMonotonicSubarray(nums: number[]): number {
  let max = 1;
  let currentLength = 1;
  const n = nums.length;

  for (let i = 1; i < n; i++) {
    if (nums[i] > nums[i - 1]) {
      currentLength++;

      max = Math.max(currentLength, max);
    } else {
      currentLength = 1;
    }
  }

  currentLength = 1;

  for (let i = 1; i < n; i++) {
    if (nums[i] < nums[i - 1]) {
      currentLength++;

      max = Math.max(currentLength, max);
    } else {
      currentLength = 1;
    }
  }

  return max;
};

// console.log(longestMonotonicSubarray([1, 4, 3, 3, 2]));
console.log(longestMonotonicSubarray([2, 1, 10]));

File: leetcode/3001-4000/3101-3200/3101-3110/3105-longest-strictly-increasing-or-strictly-decreasing-subarray/LongestStrictlyIncreasingOrStrictlyDecreasingSubarray.java
public class LongestStrictlyIncreasingOrStrictlyDecreasingSubarray {
  public static void main(String[] args) {
    System.out.println(new Solution().longestMonotonicSubarray(new int[] { 1, 2, 3, 2, 1, 0 }));
  }
}

class Solution {
  public int longestMonotonicSubarray(int[] nums) {
    int max = 1;
    int currentLength = 1;
    int n = nums.length;

    for (int i = 1; i < n; i++) {
      if (nums[i] > nums[i - 1]) {
        currentLength++;
        max = Math.max(currentLength, max);
      } else {
        currentLength = 1;
      }
    }

    currentLength = 1;

    for (int i = 1; i < n; i++) {
      if (nums[i] < nums[i - 1]) {
        currentLength++;
        max = Math.max(currentLength, max);
      } else {
        currentLength = 1;
      }
    }

    return max;
  }
}
